ibisml/steps/encode.py

```diff
@@ -152,7 +154,7 @@ def transform_table(self, table: ir.Table) -> ir.Table:
 
         return table.mutate(
             [
-                (table[col] == cat).cast("int8").name(f"{col}_{cat}")
+                ibis.ifelse((table[col] == cat), 1, 0).cast("int8").name(f"{col}_{cat}")
```

Comment:
Unnecessary cast, I think, because now you're specifying 0-1 outputs. (At least for test case, it's still `int8` output without this.)

```suggestion
                ibis.ifelse((table[col] == cat), 1, 0).name(f"{col}_{cat}")
```